lib/modules/dosomething/dosomething_reportback/dosomething_reportback.forms.inc

```diff
@@ -283,18 +286,33 @@ function dosomething_reportback_form_save_file(&$form_state) {
     $y = $form_state['values']['crop_y'];
     $width = $form_state['values']['crop_width'];
     $height = $form_state['values']['crop_height'];
+    $rotate = $form_state['values']['crop_rotate'];
     // Load the image.
     $image = image_load($file->uri);
-    // Crop the image.
-    dosomething_reportback_image_crop($image, $x, $y, $width, $height);
+    // Apply user crop and rotation to the image.
+    dosomething_reportback_image_edit($image, $x, $y, $width, $height, $rotate);
   }
   return $file;
 }
 
-// Function to crop image.
-function dosomething_reportback_image_crop($image, $x, $y, $width, $height) {
-  // Crop the image.
-  $result = image_crop($image, $x, $y, $width, $height);
+/**
+ * Rotate and crops an image based on user input.
+ */
+function dosomething_reportback_image_edit($image, $x, $y, $width, $height, $rotate) {
+  // Rotate the image and then crop it if rotation was successful.
+  if ($rotate) {
+    $result = image_rotate($image, $rotate);
```

Comment:
@DFurnes the `image_rotate()` and `image_crop` functions take and image file, apply the crop or rotation, save the image file, then returns either true or false indicating it is successful. So after it rotates, the image file is saved with the rotation, and then we load the image file again, and do the crop.